pkg/util/util.go

```diff
@@ -164,6 +173,25 @@ func GetLocalEndpoint(submSpec types.SubmarinerSpecification, backendConfig map[
 	return endpoint, nil
 }
 
+func getLabelPort(gwNode *v1.Node, labelName string) (*int32, error) {
+	var natDiscoveryPort *int32
+
+	if portStr := gwNode.Labels[labelName]; portStr != "" {
+		if portInt, err := strconv.ParseUint(portStr, 10, 16); err != nil {
+			return nil, errors.Wrapf(err, "error parsing node label port %s=%s", labelName, portStr)
+		} else if portInt < 1 {
+			return nil, errors.Errorf("node label port %s=%s is < 1", labelName, portStr)
+		} else if portInt > 65535 {
+			return nil, errors.Errorf("node label port %s=%s is > 65535", labelName, portStr)
```

Comment:
I think it would also be useful to output the node name in the messages:

```suggestion
			return nil, errors.Wrapf(err, "error parsing port label %s=%s on node %q", labelName, portStr, gwNode.Name)
		} else if portInt < 1 {
			return nil, errors.Errorf("port label %s=%s on node %q is < 1", labelName, portStr, gwNode.Name)
		} else if portInt > 65535 {
			return nil, errors.Errorf("port label %s=%s on node %q is > 65535", labelName, portStr, gwNode.Name)
```